xbmc/imagefiles/SpecialImageLoaderFactory.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "imagefiles/SpecialImageFileLoader.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+class CTexture;
+
+namespace IMAGE_FILES
+{
+class CSpecialImageLoaderFactory
+{
+public:
+  CSpecialImageLoaderFactory();
+
+  std::unique_ptr<CTexture> Load(
+      std::string specialType,
+      std::string filePath,
+      unsigned int preferredWidth,
+      unsigned int preferredHeight) const;
+
+private:
+  std::vector<std::unique_ptr<ISpecialImageFileLoader>> m_specialImageLoaders{};
```

Comment:
Since the size of this array it would be nice to use const std::array if possible.

This is only relevant if you don't plan to have some sort of registration mechanism in the future where more image loaders can by dynamically registered.

Reply:
Converted to an array. Current plans don't include dynamic registration.